wire-ios-sync-engine/Source/SessionManager/SessionManager.swift

```diff
@@ -811,9 +811,20 @@ public final class SessionManager: NSObject, SessionManagerType {
                     if error != nil {
                         self.delegate?.sessionManagerDidFailToLoadDatabase()
                     } else {
-                        let userSession = self.startBackgroundSession(for: account,
-                                                                         with: coreDataStack)
-                        completion(userSession)
+                        let userSession = self.startBackgroundSession(for: account, with: coreDataStack)
+
+                        /// If `proteusViaCoreCrypto` developer flag isOn and cryptobox directory exists,
+                        /// we need to migrate the existing proteus sessions, prekeys, and identity key to CoreCrypto keystore.
+                        let cryptoboxNeedsMigration = DeveloperFlag.proteusViaCoreCrypto.isOn && coreDataStack.cryptoboxDirectoryExists
+
+                        if cryptoboxNeedsMigration {
+                            self.delegate?.sessionManagerWillMigrateAccount {
```

Comment:
I think we can spare a part of this comment as it is literally what you wrote in code 
```suggestion
                        let cryptoboxNeedsMigration = DeveloperFlag.proteusViaCoreCrypto.isOn && coreDataStack.cryptoboxDirectoryExists

                        if cryptoboxNeedsMigration {
                            /// we need to migrate the existing proteus sessions, prekeys, and identity key to CoreCrypto keystore.

                            self.delegate?.sessionManagerWillMigrateAccount {
```

Reply:
Done